### PR TITLE
Feature: add new utility method to translate cvd values internally

### LINKED ISSUE #375: Support for Custom Value Domains
As a developer I should not have to know about custom value domains.

If I query `where=cvd=water`, I should not have to care that cvd stores the term `water` as the value `1`.

When data is displayed from this service I again should not have to know that `water` is stored as `1`. I should not have to decode the value myself, the library should handle that for me.

I recommend that this support be default but opt-out for those that know what they are doing with CVD's.

e.g.  `{decodeDomains: false}` to leave domains undisturbed.

### PR DESCRIPTION
**Pull request by jgravois**
resolves #375 (halfway anyway)
```js
// dont alter original behavior
queryFeatures({ url })
  .then(response) 

queryFeatures({ url })
    .then(queryResponse => {
      // fetch metadata and translate response
      decodeValues({ url, queryResponse })
        .then(decodedResponse)
    })


getFeatureService(url)
    .then(metadata => {
      queryFeatures({ url })
        .then(queryResponse => {
          // or pass along fields if you already have them and no internal metadata check will be issued
          decodeValues({
            url,
            queryResponse,
            fields: metadata.fields
          })
            .then(decodedResponse)
        })
    })
```

### CODE REVIEW

**Comment by jgravois** [comment]
code review has been addressed.

File: packages/arcgis-rest-feature-service/src/getFeatureService.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * const url = "https://sampleserver6.arcgisonline.com/arcgis/rest/services/ServiceRequest/FeatureServer/0";
+ *
+ * getFeatureService(url)
+ *   .then(response) // { name: "311", id: 0, ... }
+ * ```
+ *
+ * @param requestOptions - Options for the request.
+ * @returns A Promise that will resolve with the addFeatures response.
+ */
+export function getFeatureService(
```
[the remark this answers is not in the document]
Comment (jgravois) [reply packages/arcgis-rest-feature-service/src/getFeatureService.ts]: i went for parity with _another_ suggestion of yours that we've already implemented (see #122)

[the remark this answers is not in the document]
Comment (tomwayson) [reply packages/arcgis-rest-feature-service/src/getFeatureService.ts]: there is no pleasing me

**Comment by jgravois** [comment]
> swap the name `formatCodedValues()` w/ `decodeValues()`

works for me 🌴

**Pull request merged**